### PR TITLE
show(::MersenneTwister) : use module qualification

### PR DESCRIPTION
**Pull request by rfourquet**
Instead of always printing `MersenneTwister(...)`, we also prepend
`Random.` when not is scope, e.g.
```julia
julia> import Random; Random.MersenneTwister(0)
Random.MersenneTwister(0)
```

But I'm not sure whether this is the best implementation, and I don't know how to test that (as `using Random` is at the top of the test file).

### CODE REVIEW

File: stdlib/Random/src/RNGs.jl [hunk 1]
```diff
@@ -181,9 +181,9 @@ function show(io::IO, rng::MersenneTwister)
     seed = from_seed(rng.seed)
     seed_str = seed <= typemax(Int) ? string(seed) : "0x" * string(seed, base=16) # DWIM
     if rng.adv_jump == 0 && rng.adv == 0
-        return print(io, "MersenneTwister($seed_str)")
+        return print(io, "$MersenneTwister($seed_str)")
```
Comment (JeffBezanson) [inline stdlib/Random/src/RNGs.jl:184]:
```suggestion
        return print(io, MersenneTwister, "($seed_str)")
```

Comment (simeonschaub) [reply stdlib/Random/src/RNGs.jl:184]: Might as well make it
```suggestion
        return print(io, MersenneTwister, "(", seed_str, ")")
```

Comment (rfourquet) [reply stdlib/Random/src/RNGs.jl:184]: Is this only for allocation reasons?

File: stdlib/Random/src/RNGs.jl [hunk 2]
```diff
@@ -181,9 +181,9 @@ function show(io::IO, rng::MersenneTwister)
     seed = from_seed(rng.seed)
     seed_str = seed <= typemax(Int) ? string(seed) : "0x" * string(seed, base=16) # DWIM
     if rng.adv_jump == 0 && rng.adv == 0
-        return print(io, "MersenneTwister($seed_str)")
+        return print(io, MersenneTwister, "(", seed_str, ")")
     end
-    print(io, "MersenneTwister($seed_str, (")
+    print(io, "$MersenneTwister($seed_str, (")
```
Comment (rfourquet) [inline stdlib/Random/src/RNGs.jl:186]:
```suggestion
    print(io, MersenneTwister, "(", seed_str, ", (")
```

File: stdlib/Random/src/RNGs.jl (hunk 1 above)
Comment (simeonschaub) [reply stdlib/Random/src/RNGs.jl]: Yes, I would think so.

**Pull request merged**